engine/engine/src/engine.cpp

```diff
@@ -1466,294 +1425,297 @@ namespace dmEngine
+                    return;
+                }
+
+                dmInput::UpdateBinding(engine->m_GameInputBinding, dt);
+
+                engine->m_InputBuffer.SetSize(0);
+                dmInput::ForEachActive(engine->m_GameInputBinding, GOActionCallback, engine);
 
-                    dmSound::Pause(false);
+                // Sort input so that text and marked text is triggered last
+                // NOTE: Due to Korean keyboards on iOS will send a backspace sometimes to "replace" a character with a new one,
+                //       we want to make sure these keypresses arrive to the input listeners before the "new" character.
+                //       If the backspace arrive after the text, it will instead remove the new character that
+                //       actually should replace the old one.
+                qsort(engine->m_InputBuffer.Begin(), engine->m_InputBuffer.Size(), sizeof(dmGameObject::InputAction), InputBufferOrderSort);
+
+                dmArray<dmGameObject::InputAction>& input_buffer = engine->m_InputBuffer;
+                uint32_t input_buffer_size = input_buffer.Size();
+                if (input_buffer_size > 0)
+                {
+                    dmGameObject::DispatchInput(engine->m_MainCollection, &input_buffer[0], input_buffer.Size());
                 }
-            }
 
-            dmProfile::HProfile profile = dmProfile::Begin();
-            {
-                DM_PROFILE(Engine, "Frame");
 
+                dmGameObject::UpdateContext update_context;
+                update_context.m_DT = dt;
+                dmGameObject::Update(engine->m_MainCollection, &update_context);
+
+                // Don't render while iconified
+                if (!dmGraphics::GetWindowState(engine->m_GraphicsContext, dmGraphics::WINDOW_STATE_ICONIFIED))
                 {
-                    DM_PROFILE(Engine, "Sim");
-
-                    dmLiveUpdate::Update();
-                    dmResource::UpdateFactory(engine->m_Factory);
-
-                    dmHID::Update(engine->m_HidContext);
-                    if (!engine->m_RunWhileIconified) {
-                        if (dmGraphics::GetWindowState(engine->m_GraphicsContext, dmGraphics::WINDOW_STATE_ICONIFIED))
-                        {
-                            // NOTE: This is a bit ugly but os event are polled in dmHID::Update and an iOS application
-                            // might have entered background at this point and OpenGL calls are not permitted and will
-                            // crash the application
-                            dmProfile::Release(profile);
-                            return;
-                        }
-                    }
-                    /* Script context updates */
+                    // Call pre render functions for extensions, if available.
+                    // We do it here before we render rest of the frame
+                    // if any extension wants to render on under of the game.
```

Comment:
"render on under of the game" sounds a bit strange. Perhaps "render below" or "render before"